spec/controllers/password_resets_controller_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'spec_helper'
+
+describe PasswordResetsController do
+  describe "POST create" do
+    context "with blank input" do
+      it "redirects to the forgot password page" do
+        post :create, email: ''
+        expect(response).to redirect_to forgot_password_path
+      end
+
+      it "sets the error message" do
+        post :create, email: ''
+        expect(flash[:danger]).to eq("Email field cannot be blank.")
+      end
+    end
+
+    context "with email that exists in the database" do
+      it "redirects to the password reset confirmation page" do
+        Fabricate(:user, email: 'bob@example.com')
+        post :create, email: 'bob@example.com'
+        expect(response).to redirect_to password_reset_confirmation_url
+      end
+
+      it "sends out an email to the email address" do
+        Fabricate(:user, email: 'bob@example.com')
+        post :create, email: 'bob@example.com'
+        expect(ActionMailer::Base.deliveries.last.to).to eq(['bob@example.com'])
+      end
+    end
+
+    context "with email that does not exist in the database" do
+      it "redirects to the forgot password page" do
+        post :create, email: 'bob@example.com'
+        expect(response).to redirect_to forgot_password_url
+      end
+
+      it "sets the error message" do
+        post :create, email: 'bob@example.com'
+        expect(flash[:danger]).to be_present
+      end
+    end
+  end
+
+  describe "GET edit" do
+    it "renders show template if reset token is valid" do
+      alice = Fabricate(:user)
+      alice.update_column(:reset_token, '12345')
+      get :edit, id: '12345'
+      expect(response).to render_template :edit
+    end
+
+    it "redirects to the expired token page if auth token is invalid" do
+      get :edit, id: '12345'
+      expect(response).to redirect_to expired_token_path
+    end
+  end
+
+  describe "PATCH update" do
+    context "with valid reset_token" do
+      let(:alice) { Fabricate(:user, password: 'old_password') }
+
+      before do
+        alice.update_column(:reset_token, '12345')
```

Comment:
You could save a trip to the db by moving this to your fabrication. You can then change `let` to `let!` to cause the record to be created immediately, which is what I am assuming you were using this line to do.